app/controllers/api/vms_controller.rb

```diff
@@ -131,6 +131,17 @@ def rename_resource(type, id, data = {})
     end
     central_admin :rename_resource, :rename
 
+    def set_description_resource(type, id, data = {})
+      new_description = data&.dig("new_description")&.strip
+      api_resource(type, id, "Setting description for") do |vm|
+        ensure_supports(type, vm, :set_description)
+        raise BadRequestError, "Setting description requires a new_description" if new_description.blank?
+
+        task_id = vm.set_description_queue(User.current_user.userid, new_description)
+        action_result(true, "Setting description for #{model_ident(vm, type)} to #{new_description}", :task_id => task_id)
```

Comment:
do we need to tell the user what they passed in, or can we just use the default description of `"Setting description for #{model_ident(vm, type)}"` (which will get automatically added for us).